api/api.go

```diff
@@ -359,15 +360,33 @@ func (api *API) delSilence(w http.ResponseWriter, r *http.Request) {
 }
 
 func (api *API) listSilences(w http.ResponseWriter, r *http.Request) {
-	sils, err := api.silences.All()
+	var (
+		num    = r.FormValue("n")
+		offset = r.FormValue("offset")
+	)
+
+	n, err := strconv.Atoi(num)
+	if err != nil {
+		n = 50
+	}
+	o, err := strconv.Atoi(offset)
+	if err != nil {
+		o = 0
+	}
+
+	sils, err := api.silences.Query(n, o)
 	if err != nil {
 		respondError(w, apiError{
 			typ: errorInternal,
 			err: err,
 		}, nil)
 		return
 	}
-	respond(w, sils)
+
+	respond(w, map[string]interface{}{
+		"silences":      sils,
+		"totalSilences": api.silences.Count(),
```

Comment:
The silences and count get queried independently from each other with separate locks. This can result in different numbers being returned here. It should be better to return one type from `Query()` which includes both the list of requested silences as well as the total count.